src/services/helpers.ts

```diff
@@ -56,3 +58,31 @@ export const shuffleArray = <T>(array: T[]): T[] => {
   }
   return shuffled
 }
+
+export const getNextExercise = async (
+  profession: Discipline
+): Promise<{ disciplineId: number; exerciseKey: number }> => {
+  const disciplines = await loadDisciplines(profession) // TODO LUN-316 leaf disciplines must be loaded, also if nested
+  if (disciplines.length <= 0) {
+    throw new Error(`No Disciplines for id ${profession.id}`)
+  }
+  const progress = await AsyncStorage.getExerciseProgress()
+  const doneExercisesOfLeafDiscipline = (disciplineId: number): number => {
+    const progressOfDiscipline = progress[disciplineId]
+    return progressOfDiscipline
+      ? Object.keys(progressOfDiscipline).filter(item => progressOfDiscipline[item] !== undefined).length
+      : 0
+  }
+  const result = disciplines.find(discipline => doneExercisesOfLeafDiscipline(discipline.id) < exercisesWithProgress)
+  if (!result) {
+    return { disciplineId: disciplines[0].id, exerciseKey: exercisesWithoutProgress } // TODO LUN-319 show success that every exercise is done
+  }
+  const disciplineProgress = progress[result.id]
+  if (!disciplineProgress) {
+    return { disciplineId: result.id, exerciseKey: exercisesWithoutProgress }
```

Comment:
I don't really understand why you return `exercisesWithoutProgress` as `exerciseKey`. Isn't exerciseWithoutProgress just the number of exercises where no progress is saved (i.e. `1` because it is only the VocabularyList)?

Reply:
Yes you are right. But the if-condition `!disciplineProgress` gets true, if there is not progress for this discipline, therefore we want to start the next exercise, that should be done, which is the SingleChoiceExercise currently, but in general it is the next exercise after skipping the exercises that have no progress.